### PR TITLE
test: Pin API interop

### PR DESCRIPTION
**Pull request by JonKrone**
js-ipfs is close to finishing work on the pin command: https://github.com/ipfs/js-ipfs/pull/1045.

This suite of tests should give us some confidence that the go and js implementations are compatible and that, to the extent that go's pins are, js-ipfs' are correct.

Open for review 😃

### CODE REVIEW

**Comment by JonKrone** [comment]
Everything is ✔️ with the current pin work on `js-ipfs` and the latest release of go-ipfs.

I am working on testing with the current master of go-ipfs but am working through build issues :).

@dryajov jenkins CI is failing on some pubsub ascii tests on linux/windows and also on the exchange-files tests on windows. Are any of those new? Can I do something to help there?

**Comment by JonKrone** [comment]
Confirming that these tests pass with the current master of go-ipfs and the pin-api branch of js-ipfs.

File: test/pin.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+          expect(jsPins).to.deep.include.members(goPins)
+          expect(goPins).to.deep.include.members(jsPins)
+        })
+    })
+
+    // go-ipfs can read pins stored by js-ipfs
+    // tests that js' pin.flush and go's pin.load are compatible
+    it.skip('js -> go', function () {
+      // skipped because go can not be spawned on a js repo due to changes in
+      // the DataStore config [link]
```
Comment (alanshaw) [inline test/pin.js]: I'd like to understand this better...can `[link]` be a real boy?

File: package.json [hunk 2]
```diff
@@ -52,9 +52,9 @@
     "eslint-plugin-react": "^7.8.2",
     "expose-loader": "^0.7.5",
     "form-data": "^2.3.2",
-    "go-ipfs-dep": "^0.4.15",
+    "go-ipfs-dep": "~0.4.15",
     "hat": "0.0.3",
-    "ipfs": "~0.29.0",
+    "ipfs": "github:ipfs/js-ipfs",
```
Comment (alanshaw) [inline package.json]: N.B. Needs to be updated after 0.30 release

File: test/pin.js (hunk 1 above)
Comment (diasdavid) [reply test/pin.js]: LOL?

**Comment by diasdavid** [comment]
Only failure was PubSub on Windows
image

Not related to this PR.

**Pull request merged**